lib/compat/wordpress-6.5/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,639 @@
+	}
+
+	/**
+	 * Evaluates the reference path passed to a directive based on the current
+	 * store namespace, state and context.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string|true $directive_value   The directive attribute value string or `true` when it's a boolean attribute.
+	 * @param string      $default_namespace The default namespace to use if none is explicitly defined in the directive
+	 *                                       value.
+	 * @param array|false $context           The current context for evaluating the directive or false if there is no
+	 *                                       context.
+	 * @return mixed|null The result of the evaluation. Null if the reference path doesn't exist.
+	 */
+	private function evaluate( $directive_value, string $default_namespace, $context = false ) {
+		list( $ns, $path ) = $this->extract_directive_value( $directive_value, $default_namespace );
+		if ( empty( $path ) ) {
+			return null;
+		}
+
+		$store = array(
+			'state'   => isset( $this->state_data[ $ns ] ) ? $this->state_data[ $ns ] : array(),
+			'context' => isset( $context[ $ns ] ) ? $context[ $ns ] : array(),
+		);
+
+		// Checks if the reference path is preceded by a negator operator (!).
+		$should_negate_value = '!' === $path[0];
+		$path                = $should_negate_value ? substr( $path, 1 ) : $path;
+
+		// Extracts the value from the store using the reference path.
+		$path_segments = explode( '.', $path );
+		$current       = $store;
+		foreach ( $path_segments as $path_segment ) {
+			if ( isset( $current[ $path_segment ] ) ) {
+				$current = $current[ $path_segment ];
+			} else {
+				return null;
+			}
+		}
+
+		// Returns the opposite if it contains a negator operator (!).
+		return $should_negate_value ? ! $current : $current;
+	}
+
+	/**
+	 * Extracts the directive attribute name to separate and return the directive
+	 * prefix and an optional suffix.
+	 *
+	 * The suffix is the string after the first double hyphen and the prefix is
+	 * everything that comes before the suffix.
+	 *
+	 * Example:
+	 *
+	 *     extract_prefix_and_suffix( 'data-wp-interactive' )   => array( 'data-wp-interactive', null )
+	 *     extract_prefix_and_suffix( 'data-wp-bind--src' )     => array( 'data-wp-bind', 'src' )
+	 *     extract_prefix_and_suffix( 'data-wp-foo--and--bar' ) => array( 'data-wp-foo', 'and--bar' )
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $directive_name The directive attribute name.
+	 * @return array An array containing the directive prefix and optional suffix.
+	 */
+	private function extract_prefix_and_suffix( string $directive_name ): array {
+		return explode( '--', $directive_name, 2 );
+	}
+
+	/**
+	 * Parses and extracts the namespace and reference path from the given
+	 * directive attribute value.
+	 *
+	 * If the value doesn't contain an explicit namespace, it returns the default
+	 * one. If the value contains a JSON instead of a reference path, the function
```

Comment:
```suggestion
	 * one. If the value contains a JSON object instead of a reference path, the function
```

Reply:
Done in https://github.com/WordPress/gutenberg/pull/58066/commits/d4f584ae50ef07d61314a6f3395bcde6cf992ff1.